core/Pos/Binary/Class/Core.hs

```diff
@@ -1,171 +1,586 @@
-{-# LANGUAGE CPP #-}
+{-# LANGUAGE FlexibleContexts    #-}
+{-# LANGUAGE PolyKinds           #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TypeFamilies        #-}
+{-# LANGUAGE TypeOperators       #-}
+module Pos.Binary.Class.Core
+    ( Bi(..)
+    , encodeBinary
+    , decodeBinary
+    , enforceSize
+    , matchSize
+    -- * CBOR re-exports
+    , encodeListLen
+    , decodeListLen
+    , decodeListLenOf
+    , Encoding
+    , Decoder
+    -- * GHC-Generics-based encoding & decoding
+    , genericEncode
+    , genericDecode
+    ) where
 
-#include "MachDeps.h"
+import           Codec.CBOR.Decoding
+import           Codec.CBOR.Encoding
+import           GHC.Generics
```

Comment:
We have the [style-guide](https://github.com/input-output-hk/cardano-sl-style-guides/blob/master/haskell-style-guide.md#imports) which prohibits implicit imports:
> Always use explicit import lists or qualified imports. Try to use qualified imports only if import list is big enough or there are conflicts in names. This makes the code more robust against changes in these libraries.